lib/tapioca/gemfile.rb

```diff
@@ -105,15 +105,12 @@ def load_dependencies
     sig { returns([T::Enumerable[Spec], T::Array[String]]) }
     def materialize_deps
       deps = definition.locked_gems.dependencies.values
-      missing_specs = T::Array[String].new
-      materialized_dependencies = if definition.resolve.method(:materialize).arity == 1 # Support bundler >= v2.2.25
-        md = definition.resolve.materialize(deps)
-        missing_spec_names = md.missing_specs.map(&:name)
-        missing_specs = T.cast(md.missing_specs.map { |spec| "#{spec.name} (#{spec.version})" }, T::Array[String])
-        md.to_a.reject { |spec| missing_spec_names.include?(spec.name) }
-      else
-        definition.resolve.materialize(deps, missing_specs)
+      materialized_dependencies = definition.resolve.materialize(deps)
+      missing_spec_names = materialized_dependencies.missing_specs.map(&:name)
+      missing_specs = materialized_dependencies.missing_specs.map do |spec|
+        "#{spec.name} (#{spec.version})"
       end
+      materialized_dependencies.to_a.reject { |spec| missing_spec_names.include?(spec.name) }
```

Comment:
Ah, hold on, this code is not correct. It should be:
```suggestion
      materialized_dependencies.to_a.reject! { |spec| missing_spec_names.include?(spec.name) }
```